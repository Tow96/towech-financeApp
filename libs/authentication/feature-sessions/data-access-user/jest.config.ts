/* eslint-disable */
export default {
  displayName: 'authentication-feature-sessions-data-access-user',
  preset: '../../../../jest.preset.js',
  testEnvironment: 'node',
  transform: {
    '^.+\\.[tj]s$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }],
  },
  moduleFileExtensions: ['ts', 'js', 'html'],
  coverageDirectory: '../../../../coverage/libs/authentication/feature-sessions/data-access-user',
  coverageReporters: ['html', 'text', 'text-summary', 'cobertura', 'lcov', 'json'],
  collectCoverageFrom: ['**/*.ts', '!**/*.module.ts', '!**/index.ts', '!**/jest.config.ts'],
};
