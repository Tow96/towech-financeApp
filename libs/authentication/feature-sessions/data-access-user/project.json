{
  "name": "authentication-feature-sessions-data-access-user",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/authentication/feature-sessions/data-access-user/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/authentication/feature-sessions/data-access-user/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/authentication/feature-sessions/data-access-user/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    }
  },
  "tags": []
}
