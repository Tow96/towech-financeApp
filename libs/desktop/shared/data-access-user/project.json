{
  "name": "desktop-shared-data-access-user",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/desktop/shared/data-access-user/src",
  "prefix": "finance",
  "tags": [],
  "projectType": "library",
  "targets": {
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/desktop/shared/data-access-user/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/desktop/shared/data-access-user/**/*.ts",
          "libs/desktop/shared/data-access-user/**/*.html"
        ]
      }
    }
  }
}
