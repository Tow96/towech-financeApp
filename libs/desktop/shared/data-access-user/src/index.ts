export * from './lib/user.service';
export * from './lib/types';
