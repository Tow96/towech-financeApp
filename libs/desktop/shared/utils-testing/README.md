# desktop-shared-utils-testing

This library was generated with [Nx](https://nx.dev).


## Running unit tests

Run `nx test desktop-shared-utils-testing` to execute the unit tests.

