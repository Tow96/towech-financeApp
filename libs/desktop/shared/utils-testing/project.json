{
  "name": "desktop-shared-utils-testing",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/desktop/shared/utils-testing/src",
  "prefix": "finance",
  "tags": [],
  "projectType": "library",
  "targets": {
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": [
        "{workspaceRoot}/coverage/{projectRoot}"
      ],
      "options": {
        "jestConfig": "libs/desktop/shared/utils-testing/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": [
        "{options.outputFile}"
      ],
      "options": {
        "lintFilePatterns": [
          "libs/desktop/shared/utils-testing/**/*.ts",
          "libs/desktop/shared/utils-testing/**/*.html"
        ]
      }
    }
  }
}
