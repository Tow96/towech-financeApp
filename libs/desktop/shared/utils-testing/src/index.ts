export * from './lib/shared-data-access-toast.service.mock';
export * from './lib/shared-data-access-user.service.mock';
export * from './lib/shared-utils-toast.stub';
