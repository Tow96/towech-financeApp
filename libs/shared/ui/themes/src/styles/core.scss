// The theming functions are heavily taken from angular material, but in a lighter way
// Original code: https://material.angular.io/
@use 'sass:map';

@function get-palette-hue($palette, $hue) {
  @return map.get($palette, $hue);
}

@function get-palette-hue-contrast($palette, $hue) {
  @return map.get(map.get($palette, contrast), $hue);
}

@function define-theme(
  $background,
  $text,
  $primary,
  $accent,
  $success: null,
  $warning: null,
  $error: null
) {
  @if $background == null or $text == null {
    @error 'Background and text colors need to be defined';
  }

  @if $primary == null or $accent == null {
    @error 'primary and accent colors need to be defined palettes';
  }

  @if not map.get($primary, contrast) {
    @error 'Primary is not a palette';
  }

  @if not map.get($accent, contrast) {
    @error 'Accent is not a palette';
  }

  @if $success != null and not map.get($success, contrast) {
    @error 'Success is not a palette';
  }

  @if $error != null and not map.get($error, contrast) {
    @error 'Error is not a palette';
  }

  @if $warning != null and not map.get($warning, contrast) {
    @error 'Warning is not a palette';
  }

  $result: (
    color: (
      text: $text,
      background: $background,
      primary: $primary,
      accent: $accent,
      success: if($success != null, $success, $accent),
      warning: if($warning != null, $warning, $accent),
      error: if($error != null, $error, $accent),
    ),
  );

  @return $result;
}

@mixin color($theme) {
  $colors: map.get($theme, color);
  $primary: map.get($colors, primary);

  :root {
    background-color: map.get($colors, background);
    color: map.get($colors, text);
  }

  div.container {
    background-color: get-palette-hue($primary, 500);
  }
}
