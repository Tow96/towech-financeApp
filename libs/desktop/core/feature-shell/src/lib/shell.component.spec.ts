// Libraries
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { provideStore } from '@ngrx/store';
// Tested elements
import { DesktopShellComponent } from './shell.component';
// Services
import { DesktopUserService } from '@finance/desktop/shared/data-access-user';
import { DesktopUserServiceMock } from '@finance/desktop/shared/utils-testing';

describe('Desktop Shell Component', () => {
  let component: DesktopShellComponent;
  let fixture: ComponentFixture<DesktopShellComponent>;
  let compiled: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      providers: [
        provideStore(),
        { provide: DesktopUserService, useValue: DesktopUserServiceMock },
      ],
    });
    fixture = TestBed.createComponent(DesktopShellComponent);
    component = fixture.componentInstance;

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('Should be defined', () => expect(component).toBeTruthy());

  it('Must match the snapshot', () => expect(compiled).toMatchSnapshot());
});
