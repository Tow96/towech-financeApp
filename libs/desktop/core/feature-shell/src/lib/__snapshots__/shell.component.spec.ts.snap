// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop Shell Component Must match the snapshot 1`] = `
<div
  id="root1"
>
  <finance-toast-tray>
    <div
      class="desktop-toaster"
    >
      
    </div>
  </finance-toast-tray>
</div>
`;
